types/precison_timestamp_types.go

```diff
@@ -0,0 +1,167 @@
+package types
+
+import (
+	"fmt"
+	"github.com/cockroachdb/errors"
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+type TimePrecision int
+
+const (
+	UnknownPrecision TimePrecision = iota
+	Seconds
+	DeciSeconds
+	CentiSeconds
+	MilliSeconds
+	EMinus4Seconds // 10^-4 of seconds
+	EMinus5Seconds // 10^-5 of seconds
+	MicroSeconds
+	EMinus7Seconds // 10^-7 of seconds
+	EMinus8Seconds // 10^-8 of seconds
+	NanoSeconds
+)
+
+func timePrecisionToProtoVal(val TimePrecision) int32 {
```

Comment:
inclined to return error as opposed to panic.

Reply:
fixed it